cmd/apiserver-boot/boot/create_resource.go

```diff
@@ -183,6 +186,8 @@ import (
 )
 
 // +genclient=true
+{{- if .NonNamespacedKind }}
+// +nonNamespaced=true{{ end }}
```

Comment:
We still need to put the "+nonNamespaced" tag in the SecondClosestCommentLines  in order for lister-gen to generate a non-namespaced version of Lister interface.

I've found that the correct registration of a non-namespaced resource Kind requires these steps:
1. Put the "+nonNamespaced" tag in the SecondClosestCommentLines of the resource Kind.
2. Use a custom storage strategy with NamespaceScoped() returning false.
3. Provide the names of the non-namespaced resource Kinds in the GroupMetaFactoryArgs.RootScopedKinds field.
4. Change controller stub to use the non-namespaced version of the generated Lister interface.

With your recent patches, we indeed may override the default strategy by appending an extra "strategy" parameter to the "+resource" tag. But still, points 3 and 4 are located in the apiregister-gen's generated code. So I decided to reuse the already existing lister-gen's "+nonNamespaced" tag for the apiregister-gen's purposes, since its semantics is the same. But I can introduce a separate tag if it is needed.

Regarding the point 2, I've spotted your custom strategy patch too late, so I haven't rewritten the part, which puts the custom strategy into a generated file.
What do you think will be the right thing to do here?
Maybe create a xxx_strategy.go file with the custom strategy implementation stub when the "--non-namespaced" flag is present on the apiserver-boot create-resource command line?